hal/i2c_api.h

```diff
@@ -192,43 +308,28 @@ const PinMap *i2c_slave_scl_pinmap(void);
  * @{
  */
 
-/** Configure I2C as slave or master.
- *  @param obj The I2C object
- *  @param enable_slave Enable i2c hardware so you can receive events with ::i2c_slave_receive
- *  @return non-zero if a value is available
- */
-void i2c_slave_mode(i2c_t *obj, int enable_slave);
+typedef enum {
```

Comment:
Needs more doc - what's the default status? Should `NoData` be `Idle`?

Reply:
Yes, `Idle` will be more suitable here

Reply:
fixed